replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/persistence/ReplicatorConfigManagerImpl.java

```diff
@@ -0,0 +1,50 @@
+package org.codice.ditto.replication.api.impl.persistence;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.codice.ditto.replication.api.ReplicatorConfig;
+import org.codice.ditto.replication.api.impl.data.ReplicatorConfigImpl;
+import org.codice.ditto.replication.api.persistence.ReplicatorConfigManager;
+
+public class ReplicatorConfigManagerImpl implements ReplicatorConfigManager {
+
+  private ReplicationPersistentStore persistentStore;
+
+  public ReplicatorConfigManagerImpl(ReplicationPersistentStore persistentStore) {
+    this.persistentStore = persistentStore;
+  }
+
+  @Override
+  public ReplicatorConfig createConfig() {
+    return new ReplicatorConfigImpl();
+  }
+
+  @Override
+  public Optional<ReplicatorConfig> getConfig(String id) {
+    Optional<ReplicatorConfigImpl> configOptional =
+        persistentStore.get(ReplicatorConfigImpl.class, id);
+    return configOptional.isPresent() ? Optional.of(configOptional.get()) : Optional.empty();
+  }
+
+  @Override
+  public List<ReplicatorConfig> getAllConfigs() {
+    return persistentStore.objects(ReplicatorConfigImpl.class).collect(Collectors.toList());
+  }
+
+  @Override
+  public void saveConfig(ReplicatorConfig replicationConfig) {
+    if (replicationConfig instanceof ReplicatorConfigImpl) {

```

Comment:
Is this check and cast only needed because of the AbstractPersistable that the impl uses? Would be nice if we didn't have to check and cast here.

Reply:
The only way to avoid that cast is to move the `save()` directly into the `Persistable`. Then on the implementation side, you can implement without a need to check.